src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas2/ApplicationsController.kt

```diff
@@ -134,15 +136,29 @@ class ApplicationsController(
     return ResponseEntity.ok(getPersonDetailAndTransform(updatedApplication, user))
   }
 
-  private fun getPersonDetailAndTransformToSummary(
-    application: uk.gov.justice.digital
-    .hmpps.approvedpremisesapi.jpa.entity.Cas2ApplicationSummary,
-    user: NomisUserEntity,
-  ):
-    ApplicationSummary {
-    val personInfo = offenderService.getInfoForPersonOrThrowInternalServerError(application.getCrn())
+  private fun getPeopleDetailsAndTransformToSummaries(applications: List<Cas2ApplicationSummary>, user: NomisUserEntity): List<ApplicationSummary> {
+    val peopleInfo = offenderService.getOffenderSummariesByCrns(
+      applications.map { it.getCrn() }.toSet(),
+    )
+
+    val summaries = mutableListOf<ApplicationSummary>()
+
+    for (application in applications) {
+      val crn = application.getCrn()
+      val personInfo = when (val offenderDetailSummary = peopleInfo[crn]) {
+        null -> PersonInfoResult.Success.Restricted(crn, application.getNomsNumber())
```

Comment:
@Harriethw I'm not sure creating a `Restricted` PersonInfoResult is the right approach here but it allows me to cast to PersonInfoResult.Success later.